### PR TITLE
enhance: show renoters

### LINKED ISSUE #4438: その投稿をRenoteしたユーザー一覧を表示
# Summary

### PR DESCRIPTION
**Pull request by Johann150**
# What
A tooltip is added to show who has renoted a note.

I am not sure if these two additional things should be done:
- [x] a separate tooltip component can be created.
    **merit:** look cleaner.
    **demerit:** causes some duplicated code.
    Currently, the reaction component is reused with the 🔁 emoji. It looks like this:
image
- [ ] add a data field to the `api/notes/show` endpoint to be able to show when a note was renoted, without an extra API request. see https://github.com/misskey-dev/misskey/issues/4438#issuecomment-961485315
    This might even be necessary because the number of renoters is limited to `11` which might not include our own renote.
    **merit:** show user if they have shared a note before, prevent accidentall duplicate renote
    **demerit:** could cause additional database and server load

# Why
fix #4438

### CODE REVIEW

**Comment by okpierre** [comment]
I noticed your screen shot also shows duplicate emojis similar to https://github.com/misskey-dev/misskey/issues/6805 opened a year ago. Will this PR also include a fix for that?

**Comment by Johann150** [comment]
No, that is off topic.

**Comment by Johann150** [comment]
`renote.details.vue` already uses the style changes from #7942

**Comment by Johann150** [comment]
Unless you ask, I am not going to change the API here (2nd item on the checklist), because I think that can be a separate issue.

**Comment by Johann150** [comment]
updated screenshot:
image

**Comment by Johann150** [comment]
has been rebased onto the refactoring

**Review by syuilo — COMMENTED** [review]
`components/renote.vue` よりも `components/renote-button.vue` の方が分かりやすいと思います😉

File: packages/client/src/components/note.vue [hunk 1]
```diff
@@ -119,16 +114,17 @@ import XReactionsViewer from './reactions-viewer.vue';
...
+import { pleaseLogin } from '@client/scripts/please-login';
+import { focusPrev, focusNext } from '@client/scripts/focus';
+import { url } from '@client/config';
+import copyToClipboard from '@client/scripts/copy-to-clipboard';
+import { checkWordMute } from '@client/scripts/check-word-mute';
+import { userPage } from '@client/filters/user';
+import * as os from '@client/os';
+import { noteActions, noteViewInterruptors } from '@client/store';
+import { reactionPicker } from '@client/scripts/reaction-picker';
+import { extractUrlFromMfm } from '@/misc/extract-url-from-mfm';
```
Comment (syuilo) [inline packages/client/src/components/note.vue]: `@client/~`ではなく`@/~`の方が正しいです。

Comment (Johann150) [reply packages/client/src/components/note.vue]: I don't understand, I thought that was refactored?

Comment (syuilo) [reply packages/client/src/components/note.vue]: In [this commit](0e4a111f81cceed275d9bec2695f6e401fb654d8), `@client` has been replaced by `@`.
Well, I'll fix it after the merge, so no worries.

**Comment by syuilo** [comment]
🆒

File: packages/client/src/components/note.vue (hunk 1 above)
Comment (Johann150) [reply packages/client/src/components/note.vue]: Thanks ❤️

**Pull request merged**